axom@0.6.1 %oneapi build fails: (.text+0x24): undefined reference to `main'

### Steps to reproduce the issue

`axom@0.6.1 %oneapi@2022.1.0` build is failing using:
* `spack/develop` (ec879240396b457f8bf05361a7d55280e0fa19fb from `Fri Jul 29 19:12:02 2022 -0700`)
* Ubuntu 20.04, OneAPI 2022.1.0 (installed via offline installer)
* Docker container image `ecpe4s/ubuntu20.04-runner-x86_64-oneapi:2022-07-01` (OneAPI pre-installed)

This issue is coming up in our E4S OneAPI CI stack:
* https://github.com/spack/spack/blob/develop/share/spack/gitlab/cloud_pipelines/stacks/e4s-oneapi/spack.yaml#L190
* https://github.com/spack/spack/blob/develop/share/spack/gitlab/cloud_pipelines/stacks/e4s-oneapi/spack.yaml#L218

Concrete spec: [axom.spec.json.txt](https://github.com/spack/spack/files/9230409/axom.spec.json.txt)

Concretization:
```
 -   axom@0.6.1%oneapi@2022.1.0+cpp14~cuda~debug~devtools+examples+fortran+hdf5~ipo+lua~mfem+mpi+openmp~python+raja~scr+shared+tools+umpire build_type=RelWithDebInfo patches=68a3ae8 arch=linux-ubuntu20.04-x86_64
[+]      ^cmake@3.23.2%oneapi@2022.1.0~doc+ncurses+ownlibs~qt build_type=Release arch=linux-ubuntu20.04-x86_64
[+]          ^ncurses@6.2%oneapi@2022.1.0~symlinks+termlib abi=none arch=linux-ubuntu20.04-x86_64
[+]              ^pkgconf@1.8.0%oneapi@2022.1.0 arch=linux-ubuntu20.04-x86_64
[+]          ^openssl@1.1.1q%oneapi@2022.1.0~docs~shared certs=mozilla patches=3fdcf2d arch=linux-ubuntu20.04-x86_64
[+]              ^ca-certificates-mozilla@2022-03-29%oneapi@2022.1.0 arch=linux-ubuntu20.04-x86_64
[+]              ^perl@5.34.1%oneapi@2022.1.0+cpanm+shared+threads arch=linux-ubuntu20.04-x86_64
[+]                  ^berkeley-db@18.1.40%oneapi@2022.1.0+cxx~docs+stl patches=b231fcc arch=linux-ubuntu20.04-x86_64
[+]                  ^bzip2@1.0.8%oneapi@2022.1.0~debug~pic+shared arch=linux-ubuntu20.04-x86_64
[+]                      ^diffutils@3.8%oneapi@2022.1.0 arch=linux-ubuntu20.04-x86_64
[+]                          ^libiconv@1.16%oneapi@2022.1.0 libs=shared,static arch=linux-ubuntu20.04-x86_64
[+]                  ^gdbm@1.19%oneapi@2022.1.0 arch=linux-ubuntu20.04-x86_64
[+]                      ^readline@8.1.2%oneapi@2022.1.0 arch=linux-ubuntu20.04-x86_64
[+]                  ^zlib@1.2.12%oneapi@2022.1.0+optimize+pic+shared patches=0d38234 arch=linux-ubuntu20.04-x86_64
[+]      ^conduit@0.8.3%oneapi@2022.1.0~adios+blt_find_mpi~doc~doxygen+fortran+hdf5+hdf5_compat~ipo+mpi+parmetis~python+shared~silo+test~zfp build_type=RelWithDebInfo arch=linux-ubuntu20.04-x86_64
[+]          ^hdf5@1.8.21%oneapi@2022.1.0~cxx+fortran+hl~ipo+mpi+shared~szip~threadsafe+tools api=default build_type=RelWithDebInfo patches=0e20187,b61e2f0 arch=linux-ubuntu20.04-x86_64
[+]              ^mpich@4.0.2%oneapi@2022.1.0~argobots~cuda+fortran+hwloc+hydra+libxml2+pci~rocm+romio~slurm~two_level_namespace~vci~verbs~wrapperrpath datatype-engine=auto device=ch4 netmod=ofi patches=b4c706e pmi=pmi arch=linux-ubuntu20.04-x86_64
[+]                  ^findutils@4.9.0%oneapi@2022.1.0 patches=440b954 arch=linux-ubuntu20.04-x86_64
[+]                  ^hwloc@2.7.1%oneapi@2022.1.0~cairo~cuda~gl~libudev+libxml2~netloc~nvml~oneapi-level-zero~opencl+pci~rocm+shared arch=linux-ubuntu20.04-x86_64
[+]                      ^libpciaccess@0.16%oneapi@2022.1.0 arch=linux-ubuntu20.04-x86_64
[+]                          ^libtool@2.4.7%oneapi@2022.1.0 arch=linux-ubuntu20.04-x86_64
[+]                              ^m4@1.4.19%oneapi@2022.1.0+sigsegv patches=9dc5fbd,bfdffa7 arch=linux-ubuntu20.04-x86_64
[+]                                  ^libsigsegv@2.13%oneapi@2022.1.0 arch=linux-ubuntu20.04-x86_64
[+]                          ^util-macros@1.19.3%oneapi@2022.1.0 arch=linux-ubuntu20.04-x86_64
[+]                      ^libxml2@2.9.13%oneapi@2022.1.0~python arch=linux-ubuntu20.04-x86_64
[+]                          ^xz@5.2.5%oneapi@2022.1.0+pic libs=shared,static arch=linux-ubuntu20.04-x86_64
[+]                  ^libfabric@1.14.1%oneapi@2022.1.0~debug~kdreg fabrics=rxm,sockets,tcp,udp arch=linux-ubuntu20.04-x86_64
[+]                  ^yaksa@0.2%oneapi@2022.1.0~cuda~rocm arch=linux-ubuntu20.04-x86_64
[+]                      ^autoconf@2.69%oneapi@2022.1.0 patches=35c4492,7793209,a49dd5b arch=linux-ubuntu20.04-x86_64
[+]                      ^automake@1.16.5%oneapi@2022.1.0 arch=linux-ubuntu20.04-x86_64
[+]                      ^python@3.8.13%oneapi@2022.1.0+bz2+ctypes+dbm~debug+ensurepip+libxml2+lzma~nis~optimizations+pic+pyexpat+pythoncmd+readline+shared+sqlite3+ssl~tix~tkinter~ucs4+uuid+zlib patches=0d98e93,4c24573,f2fd060 arch=linux-ubuntu20.04-x86_64
[+]                          ^expat@2.4.8%oneapi@2022.1.0+libbsd arch=linux-ubuntu20.04-x86_64
[+]                              ^libbsd@0.11.5%oneapi@2022.1.0 arch=linux-ubuntu20.04-x86_64
[+]                                  ^libmd@1.0.4%oneapi@2022.1.0 arch=linux-ubuntu20.04-x86_64
[+]                          ^gettext@0.21%oneapi@2022.1.0+bzip2+curses+git~libunistring+libxml2+tar+xz arch=linux-ubuntu20.04-x86_64
[+]                              ^tar@1.34%oneapi@2022.1.0 zip=pigz arch=linux-ubuntu20.04-x86_64
[+]                                  ^pigz@2.7%oneapi@2022.1.0 arch=linux-ubuntu20.04-x86_64
[+]                                  ^zstd@1.5.2%oneapi@2022.1.0+programs compression=none libs=shared,static arch=linux-ubuntu20.04-x86_64
[+]                          ^libffi@3.4.2%oneapi@2022.1.0 arch=linux-ubuntu20.04-x86_64
[+]                          ^sqlite@3.38.5%oneapi@2022.1.0+column_metadata+dynamic_extensions+fts~functions+rtree arch=linux-ubuntu20.04-x86_64
[+]                          ^util-linux-uuid@2.37.4%oneapi@2022.1.0 arch=linux-ubuntu20.04-x86_64
[+]          ^metis@5.1.0%oneapi@2022.1.0~gdb~int64~real64+shared build_type=Release patches=4991da9 arch=linux-ubuntu20.04-x86_64
[+]          ^parmetis@4.0.3%oneapi@2022.1.0~gdb~int64~ipo+shared build_type=RelWithDebInfo patches=4f89253,50ed208,704b84f arch=linux-ubuntu20.04-x86_64
[+]      ^lua@5.3.5%oneapi@2022.1.0~pcfile+shared fetcher=curl arch=linux-ubuntu20.04-x86_64
[+]          ^curl@7.84.0%oneapi@2022.1.0~gssapi~ldap~libidn2~librtmp~libssh~libssh2~nghttp2 libs=shared,static tls=openssl arch=linux-ubuntu20.04-x86_64
[+]          ^unzip@6.0%oneapi@2022.1.0 arch=linux-ubuntu20.04-x86_64
[+]      ^raja@2022.03.0%oneapi@2022.1.0~cuda+examples+exercises~ipo+openmp~rocm+shared~tests build_type=RelWithDebInfo arch=linux-ubuntu20.04-x86_64
[+]          ^blt@0.5.1%oneapi@2022.1.0 arch=linux-ubuntu20.04-x86_64
[+]          ^camp@2022.03.0%oneapi@2022.1.0~cuda~ipo~rocm~tests build_type=RelWithDebInfo arch=linux-ubuntu20.04-x86_64
[+]      ^umpire@2022.03.1%oneapi@2022.1.0+c~cuda+device_alloc~deviceconst+examples~fortran~ipo~numa+openmp~rocm+shared build_type=RelWithDebInfo tests=none arch=linux-ubuntu20.04-x86_64
```

Install error:
```
==> Installing axom-0.6.1-ldl6uxvg7pnbmvvl5cn77synk6cmwe7q
==> No binary for axom-0.6.1-ldl6uxvg7pnbmvvl5cn77synk6cmwe7q found: installing from source
==> Using cached archive: /spack/var/spack/cache/_source-cache/git//LLNL/axom.git/v0.6.1.tar.gz
==> Warning: Fetching from mirror without a checksum!
  This package is normally checked out from a version control system, but it has been archived on a spack mirror.  This means we cannot know a checksum for the tarball in advance. Be sure that your connection to this mirror is secure!
==> Applied patch /spack/var/spack/repos/builtin/packages/axom/scr_examples_gtest.patch
==> Ran patch() for axom
==> axom: Executing phase: 'initconfig'
==> axom: Executing phase: 'cmake'
==> axom: Executing phase: 'build'
==> Error: ProcessError: Command exited with status 2:
    'make' '-j16'

5 errors found in build log:
     1255    cd /tmp/root/spack-stage/spack-stage-axom-0.6.1-ldl6uxvg7pnbmvvl5cn77synk6cmwe7q/spack-build-ldl6uxv/axom/sidre/examples && /spack/opt/spack/linux-ubuntu20.04-x86_64/oneapi-
             2022.1.0/cmake-3.23.2-4522tfkc3hkawky76rjynn3nkgkz3u2x/bin/cmake -E cmake_link_script CMakeFiles/sidre_generateindex_ex.dir/link.txt --verbose=1
     1256    /spack/lib/spack/env/oneapi/icpx      -Wall -Wextra     -O2 -g -DNDEBUG -Wl,--export-dynamic -rdynamic -fiopenmp -Wl,-rpath -Wl,/spack/opt/spack/linux-ubuntu20.04-x86_64/one
             api-2022.1.0/mpich-4.0.2-kr65wcs6yzdm6kb7kbdrg6343wwe6vye/lib CMakeFiles/sidre_generateindex_ex.dir/sidre_generateindex.cpp.o -o ../../../examples/sidre_generateindex_ex  -W
             l,-rpath,/tmp/root/spack-stage/spack-stage-axom-0.6.1-ldl6uxvg7pnbmvvl5cn77synk6cmwe7q/spack-build-ldl6uxv/lib:/spack/opt/spack/linux-ubuntu20.04-x86_64/oneapi-2022.1.0/mpic
             h-4.0.2-kr65wcs6yzdm6kb7kbdrg6343wwe6vye/lib:/spack/opt/spack/linux-ubuntu20.04-x86_64/oneapi-2022.1.0/umpire-2022.03.1-q4dfgr7l2ltn437fvid45lpclhdzg742/lib:/spack/opt/spack
             /linux-ubuntu20.04-x86_64/oneapi-2022.1.0/raja-2022.03.0-6ialib7po34zsbo4e24h3pwn6kqqw63j/lib:/spack/opt/spack/linux-ubuntu20.04-x86_64/oneapi-2022.1.0/conduit-0.8.3-ycogbd5
             vtij23gjnquu2zdgndeqr4cb6/lib:/spack/opt/spack/linux-ubuntu20.04-x86_64/oneapi-2022.1.0/hdf5-1.8.21-3cmfyzkmvc6nlnew2cfx4iccjp7blkeq/lib:/spack/opt/spack/linux-ubuntu20.04-x
             86_64/oneapi-2022.1.0/parmetis-4.0.3-osehw24exarbedyiprxvy5d4y7ugoygj/lib:/spack/opt/spack/linux-ubuntu20.04-x86_64/oneapi-2022.1.0/metis-5.1.0-izfcwp7i63zk4ti27qteddz5wnka4
             nta/lib:/spack/opt/spack/linux-ubuntu20.04-x86_64/oneapi-2022.1.0/lua-5.3.5-5i5uwrsjzkb3oo3vnowddn7ns5r6mmd2/lib ../../../lib/libaxom.so -lrt /spack/opt/spack/linux-ubuntu20
             .04-x86_64/oneapi-2022.1.0/mpich-4.0.2-kr65wcs6yzdm6kb7kbdrg6343wwe6vye/lib/libmpifort.so /spack/opt/spack/linux-ubuntu20.04-x86_64/oneapi-2022.1.0/umpire-2022.03.1-q4dfgr7l
             2ltn437fvid45lpclhdzg742/lib/libumpire.so /spack/opt/spack/linux-ubuntu20.04-x86_64/oneapi-2022.1.0/raja-2022.03.0-6ialib7po34zsbo4e24h3pwn6kqqw63j/lib/libRAJA.so /spack/opt
             /spack/linux-ubuntu20.04-x86_64/oneapi-2022.1.0/camp-2022.03.0-gwz5n6jtio2yfdejgxjjvew2ia4dfa64/lib/libcamp.a /spack/opt/spack/linux-ubuntu20.04-x86_64/oneapi-2022.1.0/condu
             it-0.8.3-ycogbd5vtij23gjnquu2zdgndeqr4cb6/lib/libconduit_relay_mpi_io.so /spack/opt/spack/linux-ubuntu20.04-x86_64/oneapi-2022.1.0/conduit-0.8.3-ycogbd5vtij23gjnquu2zdgndeqr
             4cb6/lib/libconduit_relay.so -lrt -lpthread /spack/opt/spack/linux-ubuntu20.04-x86_64/oneapi-2022.1.0/hdf5-1.8.21-3cmfyzkmvc6nlnew2cfx4iccjp7blkeq/lib/libhdf5.so.10.3.2 -ldl
              /spack/opt/spack/linux-ubuntu20.04-x86_64/oneapi-2022.1.0/conduit-0.8.3-ycogbd5vtij23gjnquu2zdgndeqr4cb6/lib/libconduit_blueprint_mpi.so /spack/opt/spack/linux-ubuntu20.04-
             x86_64/oneapi-2022.1.0/conduit-0.8.3-ycogbd5vtij23gjnquu2zdgndeqr4cb6/lib/libconduit_blueprint.so /spack/opt/spack/linux-ubuntu20.04-x86_64/oneapi-2022.1.0/conduit-0.8.3-yco
             gbd5vtij23gjnquu2zdgndeqr4cb6/lib/libconduit_relay_mpi.so /spack/opt/spack/linux-ubuntu20.04-x86_64/oneapi-2022.1.0/conduit-0.8.3-ycogbd5vtij23gjnquu2zdgndeqr4cb6/lib/libcon
             duit.so /spack/opt/spack/linux-ubuntu20.04-x86_64/oneapi-2022.1.0/mpich-4.0.2-kr65wcs6yzdm6kb7kbdrg6343wwe6vye/lib/libmpi.so /spack/opt/spack/linux-ubuntu20.04-x86_64/oneapi
             -2022.1.0/mpich-4.0.2-kr65wcs6yzdm6kb7kbdrg6343wwe6vye/lib/libmpicxx.so /spack/opt/spack/linux-ubuntu20.04-x86_64/oneapi-2022.1.0/parmetis-4.0.3-osehw24exarbedyiprxvy5d4y7ug
             oygj/lib/libparmetis.so /spack/opt/spack/linux-ubuntu20.04-x86_64/oneapi-2022.1.0/metis-5.1.0-izfcwp7i63zk4ti27qteddz5wnka4nta/lib/libmetis.so /spack/opt/spack/linux-ubuntu2
             0.04-x86_64/oneapi-2022.1.0/lua-5.3.5-5i5uwrsjzkb3oo3vnowddn7ns5r6mmd2/lib/liblua.so
     1257    [ 92%] Linking CXX executable ../../../examples/inlet_verification_ex
     1258    cd /tmp/root/spack-stage/spack-stage-axom-0.6.1-ldl6uxvg7pnbmvvl5cn77synk6cmwe7q/spack-build-ldl6uxv/axom/inlet/examples && /spack/opt/spack/linux-ubuntu20.04-x86_64/oneapi-
             2022.1.0/cmake-3.23.2-4522tfkc3hkawky76rjynn3nkgkz3u2x/bin/cmake -E cmake_link_script CMakeFiles/inlet_verification_ex.dir/link.txt --verbose=1
     1259    /spack/lib/spack/env/oneapi/icpx      -Wall -Wextra     -O2 -g -DNDEBUG -Wl,--export-dynamic -rdynamic -fiopenmp -Wl,-rpath -Wl,/spack/opt/spack/linux-ubuntu20.04-x86_64/one
             api-2022.1.0/mpich-4.0.2-kr65wcs6yzdm6kb7kbdrg6343wwe6vye/lib CMakeFiles/inlet_verification_ex.dir/verification.cpp.o -o ../../../examples/inlet_verification_ex  -Wl,-rpath,
             /tmp/root/spack-stage/spack-stage-axom-0.6.1-ldl6uxvg7pnbmvvl5cn77synk6cmwe7q/spack-build-ldl6uxv/lib:/spack/opt/spack/linux-ubuntu20.04-x86_64/oneapi-2022.1.0/umpire-2022.0
             3.1-q4dfgr7l2ltn437fvid45lpclhdzg742/lib:/spack/opt/spack/linux-ubuntu20.04-x86_64/oneapi-2022.1.0/raja-2022.03.0-6ialib7po34zsbo4e24h3pwn6kqqw63j/lib:/spack/opt/spack/linux
             -ubuntu20.04-x86_64/oneapi-2022.1.0/mpich-4.0.2-kr65wcs6yzdm6kb7kbdrg6343wwe6vye/lib:/spack/opt/spack/linux-ubuntu20.04-x86_64/oneapi-2022.1.0/conduit-0.8.3-ycogbd5vtij23gjn
             quu2zdgndeqr4cb6/lib:/spack/opt/spack/linux-ubuntu20.04-x86_64/oneapi-2022.1.0/hdf5-1.8.21-3cmfyzkmvc6nlnew2cfx4iccjp7blkeq/lib:/spack/opt/spack/linux-ubuntu20.04-x86_64/one
             api-2022.1.0/parmetis-4.0.3-osehw24exarbedyiprxvy5d4y7ugoygj/lib:/spack/opt/spack/linux-ubuntu20.04-x86_64/oneapi-2022.1.0/metis-5.1.0-izfcwp7i63zk4ti27qteddz5wnka4nta/lib:/
             spack/opt/spack/linux-ubuntu20.04-x86_64/oneapi-2022.1.0/lua-5.3.5-5i5uwrsjzkb3oo3vnowddn7ns5r6mmd2/lib ../../../lib/libaxom.so /spack/opt/spack/linux-ubuntu20.04-x86_64/one
             api-2022.1.0/umpire-2022.03.1-q4dfgr7l2ltn437fvid45lpclhdzg742/lib/libumpire.so /spack/opt/spack/linux-ubuntu20.04-x86_64/oneapi-2022.1.0/raja-2022.03.0-6ialib7po34zsbo4e24h
             3pwn6kqqw63j/lib/libRAJA.so /spack/opt/spack/linux-ubuntu20.04-x86_64/oneapi-2022.1.0/camp-2022.03.0-gwz5n6jtio2yfdejgxjjvew2ia4dfa64/lib/libcamp.a /spack/opt/spack/linux-ub
             untu20.04-x86_64/oneapi-2022.1.0/mpich-4.0.2-kr65wcs6yzdm6kb7kbdrg6343wwe6vye/lib/libmpifort.so /spack/opt/spack/linux-ubuntu20.04-x86_64/oneapi-2022.1.0/conduit-0.8.3-ycogb
             d5vtij23gjnquu2zdgndeqr4cb6/lib/libconduit_relay_mpi_io.so /spack/opt/spack/linux-ubuntu20.04-x86_64/oneapi-2022.1.0/conduit-0.8.3-ycogbd5vtij23gjnquu2zdgndeqr4cb6/lib/libco
             nduit_relay.so -lrt -lpthread /spack/opt/spack/linux-ubuntu20.04-x86_64/oneapi-2022.1.0/hdf5-1.8.21-3cmfyzkmvc6nlnew2cfx4iccjp7blkeq/lib/libhdf5.so.10.3.2 -ldl /spack/opt/sp
             ack/linux-ubuntu20.04-x86_64/oneapi-2022.1.0/conduit-0.8.3-ycogbd5vtij23gjnquu2zdgndeqr4cb6/lib/libconduit_blueprint_mpi.so /spack/opt/spack/linux-ubuntu20.04-x86_64/oneapi-
             2022.1.0/conduit-0.8.3-ycogbd5vtij23gjnquu2zdgndeqr4cb6/lib/libconduit_blueprint.so /spack/opt/spack/linux-ubuntu20.04-x86_64/oneapi-2022.1.0/conduit-0.8.3-ycogbd5vtij23gjnq
             uu2zdgndeqr4cb6/lib/libconduit_relay_mpi.so /spack/opt/spack/linux-ubuntu20.04-x86_64/oneapi-2022.1.0/conduit-0.8.3-ycogbd5vtij23gjnquu2zdgndeqr4cb6/lib/libconduit.so /spack
             /opt/spack/linux-ubuntu20.04-x86_64/oneapi-2022.1.0/mpich-4.0.2-kr65wcs6yzdm6kb7kbdrg6343wwe6vye/lib/libmpi.so /spack/opt/spack/linux-ubuntu20.04-x86_64/oneapi-2022.1.0/mpic
             h-4.0.2-kr65wcs6yzdm6kb7kbdrg6343wwe6vye/lib/libmpicxx.so /spack/opt/spack/linux-ubuntu20.04-x86_64/oneapi-2022.1.0/parmetis-4.0.3-osehw24exarbedyiprxvy5d4y7ugoygj/lib/libpa
             rmetis.so /spack/opt/spack/linux-ubuntu20.04-x86_64/oneapi-2022.1.0/metis-5.1.0-izfcwp7i63zk4ti27qteddz5wnka4nta/lib/libmetis.so /spack/opt/spack/linux-ubuntu20.04-x86_64/on
             eapi-2022.1.0/lua-5.3.5-5i5uwrsjzkb3oo3vnowddn7ns5r6mmd2/lib/liblua.so
     1260    /usr/bin/ld: /lib/x86_64-linux-gnu/crt1.o: in function `_start':
  >> 1261    (.text+0x24): undefined reference to `main'
  >> 1262    clang++: error: linker command failed with exit code 1 (use -v to see invocation)
  >> 1263    make[2]: *** [axom/quest/examples/CMakeFiles/quest_signed_distance_interface_F_ex.dir/build.make:116: examples/quest_signed_distance_interface_F_ex] Error 1
     1264    make[2]: Leaving directory '/tmp/root/spack-stage/spack-stage-axom-0.6.1-ldl6uxvg7pnbmvvl5cn77synk6cmwe7q/spack-build-ldl6uxv'
  >> 1265    make[1]: *** [CMakeFiles/Makefile2:2592: axom/quest/examples/CMakeFiles/quest_signed_distance_interface_F_ex.dir/all] Error 2
     1266    make[1]: *** Waiting for unfinished jobs....
     1267    make[2]: Leaving directory '/tmp/root/spack-stage/spack-stage-axom-0.6.1-ldl6uxvg7pnbmvvl5cn77synk6cmwe7q/spack-build-ldl6uxv'
     1268    [ 92%] Built target sidre_generateindex_ex
     1269    make[2]: Leaving directory '/tmp/root/spack-stage/spack-stage-axom-0.6.1-ldl6uxvg7pnbmvvl5cn77synk6cmwe7q/spack-build-ldl6uxv'
     1270    [ 92%] Built target inlet_verification_ex
     1271    [ 92%] Linking CXX executable ../../../examples/primal_introduction_ex
...
```

### Error message

<details><summary>Error message</summary><pre>
... see above
</pre></details>


### Information on your system

* **Spack:** 0.19.0.dev0 (ec879240396b457f8bf05361a7d55280e0fa19fb)
* **Python:** 3.8.10
* **Platform:** linux-ubuntu20.04-cascadelake
* **Concretizer:** clingo

### Additional information

[spack-build-02-cmake-out.txt](https://github.com/spack/spack/files/9230420/spack-build-02-cmake-out.txt)
[spack-build-out.txt](https://github.com/spack/spack/files/9230421/spack-build-out.txt)

@white238 @wspear @rscohn2 

### General information

- [X] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [X] I have run `spack maintainers <name-of-the-package>` and **@mentioned** any maintainers
- [X] I have uploaded the build log and environment files
- [X] I have searched the issues of this repo and believe this is not a duplicate

raja build fails for me. It might have something to do with the avx support on the machine I use to compile

```
[ 23%] Building CXX object examples/CMakeFiles/tut_atomic-histogram.dir/tut_atomic-histogram.cpp.o
cd /tmp/rscohn1/spack-stage/spack-stage-raja-2022.03.0-be6jxdcvbkhxx3u5dcmpy3pzaxkqywbf/spack-build-be6jxdc/examples && /localdisk/w\
ork/rscohn1/projects/spack/spack/lib/spack/env/oneapi/icpx  -I/tmp/rscohn1/spack-stage/spack-stage-raja-2022.03.0-be6jxdcvbkhxx3u5dc\
mpy3pzaxkqywbf/spack-src/include -I/tmp/rscohn1/spack-stage/spack-stage-raja-2022.03.0-be6jxdcvbkhxx3u5dcmpy3pzaxkqywbf/spack-build-\
be6jxdc/include -isystem /localdisk/work/rscohn1/projects/spack/spack/opt/spack/linux-ubuntu20.04-cascadelake/oneapi-2022.1.0/camp-2\
022.03.0-sobu6mf4fnpznld3ffsk7nktodlamz24/include -Wall -Wextra      -O2 -g -DNDEBUG -fPIE -fiopenmp -std=c++17 -MD -MT examples/CMa\
keFiles/tut_scan.dir/tut_scan.cpp.o -MF CMakeFiles/tut_scan.dir/tut_scan.cpp.o.d -o CMakeFiles/tut_scan.dir/tut_scan.cpp.o -c /tmp/r\
scohn1/spack-stage/spack-stage-raja-2022.03.0-be6jxdcvbkhxx3u5dcmpy3pzaxkqywbf/spack-src/examples/tut_scan.cpp
cd /tmp/rscohn1/spack-stage/spack-stage-raja-2022.03.0-be6jxdcvbkhxx3u5dcmpy3pzaxkqywbf/spack-build-be6jxdc/examples && /localdisk/w\
ork/rscohn1/projects/spack/spack/lib/spack/env/oneapi/icpx  -I/tmp/rscohn1/spack-stage/spack-stage-raja-2022.03.0-be6jxdcvbkhxx3u5dc\
mpy3pzaxkqywbf/spack-src/include -I/tmp/rscohn1/spack-stage/spack-stage-raja-2022.03.0-be6jxdcvbkhxx3u5dcmpy3pzaxkqywbf/spack-build-\
be6jxdc/include -isystem /localdisk/work/rscohn1/projects/spack/spack/opt/spack/linux-ubuntu20.04-cascadelake/oneapi-2022.1.0/camp-2\
022.03.0-sobu6mf4fnpznld3ffsk7nktodlamz24/include -Wall -Wextra      -O2 -g -DNDEBUG -fPIE -fiopenmp -std=c++17 -MD -MT examples/CMa\
keFiles/tut_atomic-histogram.dir/tut_atomic-histogram.cpp.o -MF CMakeFiles/tut_atomic-histogram.dir/tut_atomic-histogram.cpp.o.d -o \
CMakeFiles/tut_atomic-histogram.dir/tut_atomic-histogram.cpp.o -c /tmp/rscohn1/spack-stage/spack-stage-raja-2022.03.0-be6jxdcvbkhxx3\
u5dcmpy3pzaxkqywbf/spack-src/examples/tut_atomic-histogram.cpp
In file included from /tmp/rscohn1/spack-stage/spack-stage-raja-2022.03.0-be6jxdcvbkhxx3u5dcmpy3pzaxkqywbf/spack-src/examples/tut_da\
xpy.cpp:12:
In file included from /tmp/rscohn1/spack-stage/spack-stage-raja-2022.03.0-be6jxdcvbkhxx3u5dcmpy3pzaxkqywbf/spack-src/include/RAJA/RA\
JA.hpp:67:
In file included from /tmp/rscohn1/spack-stage/spack-stage-raja-2022.03.0-be6jxdcvbkhxx3u5dcmpy3pzaxkqywbf/spack-src/include/RAJA/po\
licy/tensor.hpp:23:
In file included from /tmp/rscohn1/spack-stage/spack-stage-raja-2022.03.0-be6jxdcvbkhxx3u5dcmpy3pzaxkqywbf/spack-src/include/RAJA/po\
licy/tensor/arch_impl.hpp:35:
In file included from /tmp/rscohn1/spack-stage/spack-stage-raja-2022.03.0-be6jxdcvbkhxx3u5dcmpy3pzaxkqywbf/spack-src/include/RAJA/po\
licy/tensor/arch/avx512.hpp:22:
/tmp/rscohn1/spack-stage/spack-stage-raja-2022.03.0-be6jxdcvbkhxx3u5dcmpy3pzaxkqywbf/spack-src/include/RAJA/policy/tensor/arch/avx51\
2/avx512_int32.hpp:213:5: error: C-style cast from vector 'RAJA::expt::Register<int, RAJA::expt::avx512_register>::register_type' (a\
ka '__m512i') to vector '__m256i' (vector of 4 'long long' values) of different size
                                _mm512_i64scatter_epi32(ptr,
                                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

--- a/share/spack/gitlab/cloud_pipelines/stacks/e4s-oneapi/spack.yaml
+++ b/share/spack/gitlab/cloud_pipelines/stacks/e4s-oneapi/spack.yaml
@@ -100,6 +100,7 @@ spack:
   - amrex
   - arborx
   - argobots
+  - axom
   - bolt
   - bricks
   - butterflypack
@@ -190,7 +191,6 @@ spack:
   #- adios2@2.8.0                                     # adios2
   #- archer@2.0.0                                     # binutils
   #- ascent ^vtk-m ~openmp                            # ascent
-  #- axom@0.6.1                                       # axom
   #- charliecloud@0.26                                # charliecloud
   #- dyninst@12.1.0                                   # intel-tbb
   #- exaworks@0.1.0                                   # rust, flux-sched
@@ -216,7 +216,6 @@ spack:
   # CPU BUILD FAILURES - NOTES
   # adios2: /usr/bin/ld: ../../lib/libadios2_fortran.so.2.8.2: version node not found for symbol adios2_adios_init_mod@adios2_adios_init_serial_smod._; /usr/bin/ld: failed to set dynamic section sizes: bad value
   # ascent: examples/proxies/cloverleaf3d-ref/clover_main.cpp:24: multiple definition of `main'; /opt/intel/oneapi/compiler/2022.1.0/linux/compiler/lib/intel64_lin/for_main.o:for_main.c:(.text+0x0): first defined here
-  # axom: /usr/bin/ld: /lib/x86_64-linux-gnu/crt1.o: in function `_start': (.text+0x24): undefined reference to `main'
   # binutils: gold/powerpc.cc:3590: undefined reference to `gold::Sized_symbol<64>::Value_type gold::Symbol_table::compute_final_value<64>(gold::Sized_symbol<64> const*, gold::Symbol_table::Compute_final_value_status*) const'
   # boost@1.79.0 cxxstd=17: clang++: error: unsupported argument 'h-create' to option '-pc'; clang++: error: no such file or directory: 'bin.v2/libs/math/build/intel-linux-2022.1.0/release/cxxstd-17-iso/threading-multi/visibility-hidden/pch.pchi'
   # charliecloud: autoreconf phase: RuntimeError: configure script not found in ...

--- a/var/spack/repos/builtin/packages/axom/examples-oneapi.patch
+++ b/var/spack/repos/builtin/packages/axom/examples-oneapi.patch
@@ -1,0 +1,12 @@
+diff -ruN spack-src/src/axom/quest/examples/CMakeLists.txt spack-src-patched/src/axom/quest/examples/CMakeLists.txt
+--- spack-src/src/axom/quest/examples/CMakeLists.txt	2022-08-08 08:51:26.746637415 -0700
++++ spack-src-patched/src/axom/quest/examples/CMakeLists.txt	2022-08-08 08:52:26.238959054 -0700
+@@ -103,7 +103,7 @@
+     # When CUDA is enabled, BLT will determine the correct linker, so don't override it here
+     if (NOT ENABLE_CUDA)
+         # When using the Intel compiler we need to link with the Fortran compiler to get openmp to work correctly.
+-        if ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Intel")
++	if ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Intel" OR "${CMAKE_CXX_COMPILER_ID}" STREQUAL "IntelLLVM")
+             set_target_properties( quest_signed_distance_interface_F_ex
+                                    PROPERTIES LINKER_LANGUAGE Fortran)
+ 

--- a/var/spack/repos/builtin/packages/axom/package.py
+++ b/var/spack/repos/builtin/packages/axom/package.py
@@ -54,6 +54,9 @@ class Axom(CachedCMakePackage, CudaPackage):
     def submodules(self):
         return True
 
+    # https://github.com/spack/spack/issues/31829
+    patch("examples-oneapi.patch", when="@0.6.1 +examples %oneapi")
+
     patch("scr_examples_gtest.patch", when="@0.6.0:0.6.1")
 
     root_cmakelists_dir = "src"

